### ISSUE TITLE
This is not very flexible isn't it? this should be extracted....

### ISSUE THREAD

**Issue by marcelopereiraarxi** [odoo/odoo#90994]
https://github.com/odoo/odoo/blob/ee7a8ce740d1ec1ce6bdc0e992b8fee4fa975bdb/addons/hr_timesheet/models/hr_timesheet.py#L145

**Comment by gdgellatly** [comment]
The su cutout should already be enough shouldn't it? Makes it very easy to override.

**Comment by marcelopereiraarxi** [comment]
so you are suggesting that in order to override this, override the write, do our own validation, and then run the super with sudo, is better than extract this to a method (like in every single other place in odoo) and call this method in the write?

**Comment by gdgellatly** [comment]
Lol, every single other place in Odoo? You have obviously had a very pain free Odoo life. We have to do crap like this all the time. Add context keys, rebrowse returns, overwrite functions, monkey patch all sorts. This thing is a piece of cake, exactly do your validation, and if it is valid, then su away just for the super call.  You will never get it changed in stable, so you can complain, or actually make it work for you.

**Comment by marcelopereiraarxi** [comment]
we are talking about a validation done in the write, that should be done with an ir.rule

**Comment by gdgellatly** [comment]
I disagree. I find ir.rule so problematic in these situations because of how it chains logic. The GLOBAL AND (x or y) AND (x or y) just gives you weird results too often.

Basically impossible to get that exact result and hope it works with the preexisting one and/or future ones. I find they really only work well if there is 1 and only 1 rule per group per field per model.

But in any case, as I say, no one is going to fix it in stable, so best deal with it.
